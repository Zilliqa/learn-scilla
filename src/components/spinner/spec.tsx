import React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Spinner from '.';

describe('Spinner tests', () => {
  const baseComponent = () => <Spinner />;

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = renderer.create(baseComponent()).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders the component', () => {
      const wrapper = shallow(baseComponent());
      const assertion = wrapper.find('[data-test-id="spinner"]').length;
      expect(assertion).toBe(1);
    });
  });
});
