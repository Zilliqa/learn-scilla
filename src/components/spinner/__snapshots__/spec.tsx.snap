// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner tests basic tests matches the snapshot 1`] = `
<div
  className="py-5 text-center"
  data-test-id="spinner"
>
  <div
    className="spinner"
  />
</div>
`;
